### PR TITLE
Feature setup stripe

### PR DESCRIPTION
**Pull request by sagar-13**
- Chefs stripe onboarding, redirect on registration, login links, etc 
- Checkout form with cart summary and credit integration
- Payment goes through (no post-payment fulfillment at the moment but it will show on stripe dashboard)
- Implemented snackbar context and added corresponding alerts to every page that would benefit from alerts (The snackbars will persist across screen changes now)

### CODE REVIEW

File: client/src/components/SetUpStripe.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+// import { useHistory } from "react-router-dom";
```
Comment (ThinkTink) [inline client/src/components/SetUpStripe.js]:
```suggestion
```

File: client/src/components/SetUpStripe.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+import { Typography, Grid, Button } from "@material-ui/core";
+// import { makeStyles } from "@material-ui/core/styles";
```
Comment (ThinkTink) [inline client/src/components/SetUpStripe.js]:
```suggestion
```

File: client/src/components/SetUpStripe.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    const [account, setAccount] = React.useState([]);
+
+    React.useEffect(() => {
+        let mounted = true;
+        user.getStripeAccount().then(
+            (data) => {
+                if (mounted) {
+                    setIsLoaded(true);
+                    setAccount(data);
+                    console.log(data);
```
Comment (ThinkTink) [inline client/src/components/SetUpStripe.js]:
```suggestion
```

File: client/src/index.js [hunk 4]
```diff
@@ -3,11 +3,19 @@ import ReactDOM from "react-dom";
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
 import { UserContextProvider } from "./contexts/user/UserContextProvider";
+import { Elements } from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+
+const stripePromise = loadStripe(
+    "pk_test_51HrOLODpMZzy7YFf2ImPzBZIAj7wyO69oVlA2hyu7e7BuLctRAzEVxTAnLwkRmKHdUgLW4SRjPpnPvxJyJMAumHq00PNeexRAM"
```
Comment (ThinkTink) [inline client/src/index.js:10]: Should this be in a .env?

File: client/src/pages/Checkout.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    ListItemSecondaryAction,
+    ListItemText,
+    Typography,
+} from "@material-ui/core";
+
+import DeleteIcon from "@material-ui/icons/Delete";
+
+import { CartContext } from "../contexts/cart/CartContextProvider";
+import { dollarFormatter, calcServiceFee, calcTotalWithFee } from "../lib/utils";
+// import { useHistory } from "react-router-dom";
```
Comment (ThinkTink) [inline client/src/pages/Checkout.js]:
```suggestion
```

File: client/src/pages/Checkout.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    quantityBttnMinus: {
+        minWidth: "43px",
+        padding: 0,
+        margin: 0,
+    },
+}));
+
+function Checkout(props) {
+    const classes = useStyles();
+    // const history = useHistory();
```
Comment (ThinkTink) [inline client/src/pages/Checkout.js]:
```suggestion
```

File: client/src/pages/Checkout.js [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+    const {
+        // chefName,
```
Comment (ThinkTink) [inline client/src/pages/Checkout.js]:
```suggestion
```

File: server/routes/auth.js [hunk 8]
```diff
@@ -84,22 +81,38 @@ router.post(
             [street, city, region, country] = addressArr;
         }
 
+        // STRIPE CONNECTED ACCOUNT ONBOARDING FOR CHEFS
+        var stripeId = "";
```
Comment (ThinkTink) [inline server/routes/auth.js]: * let

File: server/routes/auth.js [hunk 9]
```diff
@@ -84,22 +81,38 @@ router.post(
...
             email,
             password,
             isChef,
+            stripeId,
         });
 
         const responseObj = await createAuthResponseObj(createdUser);
         res.cookie("token", responseObj.token, { httpOnly: true });
         res.status(201).json(responseObj);
+        // res.redirect("../chefs/stripe");
```
Comment (ThinkTink) [inline server/routes/auth.js]:
```suggestion
```

**Review by ThinkTink — APPROVED** [review]
This works for now!
I'm not a fan of using window.location.replace, so in the future we can look at alternatives to that + the other clean up you've identified

File: client/src/index.js (hunk 4 above)
Comment (sagar-13) [reply client/src/index.js:10]: Yes, it probably should. I will add that in a future commit (Should be safe for now since it's only a test key)

**Pull request merged**